tests/common/include/tests/common/test_toy_detector.hpp

```diff
@@ -111,20 +83,24 @@ inline bool test_toy_detector(
 
     // Link to outer world (leaving detector)
     constexpr auto leaving_world{detail::invalid_value<nav_link_t>()};
+    const bool has_grids =
+        (sf_finders.size<sf_finder_ids::e_cylinder_grid>() != 0u) ||
+        (sf_finders.size<sf_finder_ids::e_disc_grid>() != 0u);
 
     // Check number of geomtery objects
     EXPECT_EQ(volumes.size(), 20u);
-    EXPECT_EQ(surfaces.size(), 3244u);
-    /*EXPECT_EQ(sf_finders.template size<sf_finder_ids::e_brute_force>(), 1);
-    EXPECT_EQ(sf_finders.template size<sf_finder_ids::e_cylinder_grid>(),
-              n_brl_layers);
-    EXPECT_EQ(sf_finders.template size<sf_finder_ids::e_disc_grid>(), 14);*/
+    EXPECT_EQ(toy_det.n_surfaces(), 3244u);
     EXPECT_EQ(transforms.size(ctx), 3244u);
-    EXPECT_EQ(masks.template size<mask_ids::e_rectangle2>(), 2492u);
-    EXPECT_EQ(masks.template size<mask_ids::e_trapezoid2>(), 648u);
-    EXPECT_EQ(masks.template size<mask_ids::e_portal_cylinder2>(), 52u);
-    EXPECT_EQ(masks.template size<mask_ids::e_portal_ring2>(), 52u);
-    // EXPECT_EQ(materials.template size<material_ids::e_slab>(), 3244u);
+    EXPECT_EQ(masks.size<mask_ids::e_rectangle2>(), 2492u);
+    EXPECT_EQ(masks.size<mask_ids::e_trapezoid2>(), 648u);
+    EXPECT_EQ(masks.size<mask_ids::e_portal_cylinder2>(), 52u);
+    EXPECT_EQ(masks.size<mask_ids::e_portal_ring2>(), 52u);
+    EXPECT_EQ(sf_finders.size<sf_finder_ids::e_brute_force>(), 20u);
+    if (has_grids) {
```

Comment:
The detector writer cannot write grids, yet, so when we make the IO rountrip test, no grids are read in. In that case, skip the grid test for now